CMakeLists.txt

```diff
@@ -219,6 +219,16 @@ else()
   add_cxx_compiler_flag(--coverage COVERAGE)
 endif()
 
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Intel" AND "${CMAKE_BUILD_TYPE}" STREQUAL "Release")
+	# Handle some issues: 
+	# 1. implicit type casting not allowed (sysinfo.cc)
+    add_cxx_compiler_flag(-wd2259)
+    # 2. Function declared but never referenced (sysinfo.cc)
+    add_cxx_compiler_flag(-wd177)    
+    # 3. Unrecognized #pragma (gtest-port.cc)
```

Comment:
this is googletest issue

Reply:
agreed. but given it affects us is it a problem to disable this warning?

Reply:
not really, but this should ideally reference the upstream issue/pr with the same diff